crux-agent/app/core/orchestrators/enhanced.py

```
@@ -187,16 +195,26 @@ def professor_progress(current_iter: int, max_iters: int, phase: str):
         
         # Extract specialist consultation results from Professor's evolution history
         specialist_results = []
+        seen_results = set()  # Deduplicate across history vs final metadata
         total_specialist_tokens = 0
         total_reasoning_tokens = 0
+        context_summarized_any = bool(professor_solution.metadata.get("context_summarized", False))
+        context_truncated_any = bool(professor_solution.metadata.get("context_truncated", False))
         
```

Ruff (0.12.2)

Blank line contains whitespace

Remove whitespace from blank line

(W293)